### ISSUE TITLE
Describe the file size limitations better

### ISSUE THREAD

**Issue by Misiu** [microsoftgraph/microsoft-graph-docs#6555]
Currently, there is just info that if the file size is under 4MB we can use the old method, when the file size is between 3-150MB we must create an upload session.
What should we do if we want to send multiple files, for example, five 1MB pdf's? We can't use the old method because the limit is exceeded, we can't use the new method because the single file is just 1MB.
What is the solution?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 60e6b4ed-4a94-6789-95e3-407780e8fd7e
* Version Independent ID: ff8aca6f-1a1f-1bef-c1dd-bfdbbaf9f901
* Content: [Attach large files to Outlook messages - Microsoft Graph](https://docs.microsoft.com/en-us/graph/outlook-large-attachments?tabs=csharp)
* Content Source: [concepts/outlook-large-attachments.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/concepts/outlook-large-attachments.md)
* Product: **outlook**
* Technology: **microsoft-graph**
* GitHub Login: @angelgolfer-ms
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by angelgolfer-ms** [comment]
@svpsiva , would you be able to recommend a solution to @Misiu ?
If appropriate, then either you or I can update the docs for clarification.

**Comment by Misiu** [comment]
@angelgolfer-ms @svpsiva any updates? This is an important issue. Zipping all files into one isn't an option.

**Comment by darrelmiller** [comment]
Why can't you use the "new method" with 1MB files?

**Comment by Misiu** [comment]
@darrelmiller because I'm not allowed.
According to the docs:

> Depending on the size of the file, you can choose one of two ways to attach a file to a message:
> 
> If the file size is under 4 MB, you can do a single POST on the attachments navigation property of the message. The successful POST response includes the ID of the file attached to the message.

> If the file size is between 3MB and 150MB, create an upload session, and iteratively use PUT to upload ranges of bytes of the file until you have uploaded the entire file. A header in the final successful PUT response includes a URL with the attachment ID.

I've tried creating an upload session for 1MB file but I got an error about file size. I'll recreate the error and post it here.

**Comment by darrelmiller** [comment]
@Misiu Ok, that surprises me.  @nimeesh-msft What was the reason for preventing creating an UploadSession for files < 4MB?

@Misiu I'm also a bit confused.  Could you summarize the calls you are making and when you are getting the error message?

**Comment by Misiu** [comment]
@darrelmiller I was able to recreate the error.
When I try to create upload session for a file that has size 759 KB I get following error:

```
{
  "error": {
    "code": "ErrorAttachmentSizeShouldNotBeLessThanMinimumSize",
    "message": "Attachment size must be greater than the minimum size.",
    "innerError": {
      "request-id": "6f1dce51-9f41-4baa-9d8c-aedded540ab0",
      "date": "2020-01-03T08:22:33"
    }
  }
}
```

When I try to send a file that has 10 MB size everything works as expected.

Below is code from my test app:

```
var confidentialClientApplication = ConfidentialClientApplicationBuilder.Create(clientId).WithClientSecret(clientSecret).WithTenantId(tenant).Build();
var authenticationProvider = new ClientCredentialProvider(confidentialClientApplication);
Beta.IGraphServiceClient graphClient = new Beta.GraphServiceClient(authenticationProvider);

var recipientList = new List<Beta.Recipient>
{
    new Beta.Recipient {EmailAddress = new Beta.EmailAddress {Address = "test@example.com"}}
};

var email = new Beta.Message
{
    Body = new Beta.ItemBody
    {
        Content = "Works fine",
        ContentType = Beta.BodyType.Html,
    },
    Subject = "Test",
    ToRecipients = recipientList,
};

Beta.Message draft = await graphClient.Users["a@b.com"].MailFolders.Drafts.Messages.Request().WithImmutableId().WithMaxRetry(5).AddAsync(email);

var stream = File.Open("D:\\10MB.zip", System.IO.FileMode.Open, FileAccess.Read, FileShare.None);

var attachmentItem = new Beta.AttachmentItem
{
    AttachmentType = Beta.AttachmentType.File,
    Name = "10MB.zip",
    Size = stream.Length
};

var uploadSession = await graphClient.Users["a@b.com"].Messages[draft.Id].Attachments
    .CreateUploadSession(attachmentItem)
    .Request()
    .PostAsync();

var maxSliceSize = 320 * 1024; // 320 KB - Change this to your slice size. 5MB is the default.
var largeFileUploadTask = new LargeFileUploadTask<FileAttachment>(uploadSession, stream, maxSliceSize);

// upload away with relevant callback
try
{
    var progressCallback = new ProgressCallback(stream.Length);
    var uploadResult = await largeFileUploadTask.UploadAsync(progressCallback);
    if (uploadResult.UploadSucceeded)
    {
        await graphClient.Users["a@b.com"].Messages[draft.Id].Send().Request().WithImmutableId().WithMaxRetry(5).PostAsync();
    }
}
catch (ServiceException exx)
{
    Console.WriteLine(exx.Message);
}
```

With such limitations, I'm unable to send an email with a couple of attachments (as described in the first post).

@darrelmiller @nimeesh-msft any workarounds? I think this is a huge limitation and/or a bug.

**Comment by Misiu** [comment]
@darrelmiller any updates on this?

**Comment by keithdisarno** [comment]
Leaving a comment to hopefully get notified. I'm in the same spot at @Misiu.

I find it hard to believe someone, somewhere typed "ErrorAttachmentSizeShouldNotBeLessThanMinimumSize" into an IDE, committed it, it was reviewed, and pushed out....without thinking about anything less than 4MB. They've completely fragmented adding attachments at best (those above 4MB use session, those below use sendMail), and at worst, broke it completely in situations like OP's.

Actually...it may not be that hard to believe. The whole reason we're here is because of a (impossible to change?!?!) 4MB limit on the sendMail endpoint. I don't agree with how @maitra said it, but they were spot on. https://github.com/microsoftgraph/microsoft-graph-docs/issues/4370#issuecomment-511063026

**Comment by mhagesfeld** [comment]
Following...other than zipping or sending multiple messages, is there any sort of workaround?

**Comment by nrazon** [comment]
any new updates about this issue? And also any ETA for releasing this api method to 1.0 from beta!?

**Comment by Misiu** [comment]
@nrazon I hope this won't get released until this issue is fixed

**Comment by svpsiva** [comment]
Sorry for jumping on this so late. Did you try using the following step? 

If the file size is under 4 MB, you can do a single POST on the attachments navigation property of the message. The successful POST response includes the ID of the file attached to the message. 

https://docs.microsoft.com/en-us/graph/api/message-post-attachments?view=graph-rest-beta&tabs=http

For five 1MB files, you will need to make 5 requests to the POST /me/messages/{id}/attachments

**Comment by Misiu** [comment]
@svpsiva I found this in docs:

> Note: This operation limits the size of the attachment you can add to under 4 MB.
> 
> However, if you're attaching to a message a file that is between 3MB and 150MB, you can create an upload session and iteratively upload ranges of the file to attach it. See attach large files to Outlook messages for an example.

does this mean that I can use both methods for files that have a size between 3 and 4 MB?

**Comment by svpsiva** [comment]
As the base64 encoded filesize could be > 4MB for a 3.5 MB file, uploads of such files might fail with the old non-UploadSession approach. For this reason, we suggest the use of UploadSessions for all files with size > 3MB.

**Comment by Misiu** [comment]
@svpsiva thank you for the reply.
Maybe a good idea would be to update the docs, as it is a bit confusing right now.

**Comment by mhagesfeld** [comment]
Thank you @svpsiva!  I feel like I was running into a wall repeatedly, and you showed me the door through :)  Can't believe I didn't think of that, thanks so much!

**Comment by angelgolfer-ms** [comment]
Thank you everyone for your patience! FYI @svpsiva updated the docs in PR #7044 , and the updates will go live within the next day.
